network/transport/v2/protocol.proto

```diff
@@ -27,10 +27,27 @@ service Protocol {
 }
 
 message Message {
-  oneof message {
-      HelloMessage hello = 1000;
-  }
+
+  repeated HelloMessage hello = 1000;
+
+  repeated TransactionPayloadQuery transactionPayloadQuery = 100;
+  repeated TransactionPayload transactionPayload = 101;
```

Comment:
Why aren't we using `oneof` here? It makes processing the message ugly as we have to iterate through all fields manually to get the specific message..

Reply:
I decided it would be nice to be able to pack multiple messages in a single envelope. But as long as we don't use it, we could just revert it back to being `oneof`. Might be more simpler to read.

Reply:
> I don't think there are particular disadvantages on encoding/decoding/network layer to allowing multiple messages to be packed.

I agree. The only advantage is that it's nice to be able to do a type switch on the message when using `oneof`. If we do want to pack multiple messages in a single envelope we could create a `kind` field on the `Message` which is a `repeated MessageKind` where `MessageKind` is using `oneof`.

Reply:
Ok I will revert to simple `oneoff`